src/transformers/modeling_utils.py

```diff
@@ -4324,6 +4357,14 @@ def from_pretrained(
                 }
             return model, loading_info
 
+        if tp_plan is not None:
+            assert tp_device is not None, "tp_device not set!"
```

Comment:
let's raise n error instead

Reply:
Oh, here the assert is more for catching internal error than user error.
It is like "I swear this cannot happen” but it seems to happen anyway.
If someone runs python with `-O` flag, this assert can be optimized away :)

Reply:
Okay!